### PR TITLE
Basic support for POCO mappings

### PR DESCRIPTION
**Pull request by asarnaout**

### CODE REVIEW

File: RediSearchClient.SampleData/Program.cs [hunk 1]
```diff
@@ -5,8 +5,9 @@
 using System;
 using System.Linq;
 using NReJSON;
+using RediSearchClient.SampleData.TestingTypes;
 
-using var muxr = ConnectionMultiplexer.Connect("localhost");
+using var muxr = ConnectionMultiplexer.Connect("192.168.0.59:7000,192.168.0.57:7001,192.168.0.59:7002,192.168.0.59:7003,192.168.0.57:7004,192.168.0.59:7005");
```
Comment (tombatron) [inline RediSearchClient.SampleData/Program.cs:10]: Let's not change this. It's probably safe to assume that the basic use case here is a single node local instance.

File: RediSearchClient/Converters/BoolToStringConverter.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace RediSearchClient.Converters
+{
+    /// <summary>
+    /// RediSearch does not natively support boolean types, therefore we need to convert booleans to strings in this case
+    /// </summary>
+    internal sealed class BoolToStringConverter : JsonConverter<bool>
```
Comment (asarnaout) [inline RediSearchClient/Converters/BoolToStringConverter.cs:10]: I actually haven't tried out indexing booleans as tags yet, might resolve this issue and we dont need that converter

File: RediSearchClient.SampleData/Program.cs (hunk 1 above)
Comment (asarnaout) [reply RediSearchClient.SampleData/Program.cs:10]: Yep, will undo that change

File: RediSearchClient/Converters/BoolToStringConverter.cs (hunk 2 above)
Comment (asarnaout) [reply RediSearchClient/Converters/BoolToStringConverter.cs]: Removed as Tags are better suited to handle booleans

File: RediSearchClient.SampleData/Program.cs (hunk 1 above)
Comment (asarnaout) [reply RediSearchClient.SampleData/Program.cs]: Fixed

**Pull request closed without merging**